ir/src/lib.rs

```diff
@@ -75,18 +84,26 @@ impl AirIR {
                     for constraint in constraints.boundary_constraints.iter() {
                         boundary_constraints.insert(&symbol_table, constraint)?;
                     }
+                    validator.exists("boundary_constraints");
                 }
                 ast::SourceSection::TransitionConstraints(constraints) => {
                     for constraint in constraints.transition_constraints.iter() {
                         transition_constraints.insert(&symbol_table, constraint)?;
                     }
+                    validator.exists("transition_constraints");
                 }
                 _ => {}
             }
         }
 
         let (public_inputs, periodic_columns) = symbol_table.into_declarations();
 
+        // validate sections
+        validator.check("trace_columns")?;
+        validator.check("public_inputs")?;
+        validator.check("boundary_constraints")?;
+        validator.check("transition_constraints")?;
+
```

Comment:
I think we should handle this as one call `validator.check()` so that when we have other things that need to be processed and then validated in the future it can all be captured in the same call